src/domain-services/plans/plan-service.ts

```diff
@@ -1,42 +1,51 @@
-import {
-  PlanVersionTable,
-  PlanYearTable,
-} from '../../data-providers/postgres/models/plans/plan';
-import { YearTable } from '../../data-providers/postgres/models/year';
+import { PlanId } from '@unocha/hpc-api-core/src/db/models/plan';
+import { Database } from '@unocha/hpc-api-core/src/db/type';
+import { createBrandedValue } from '@unocha/hpc-api-core/src/util/types';
 import { Service } from 'typedi';
-import { DbModels } from '../../data-providers/postgres/models';
 
 @Service()
 export class PlanService {
-  async findById(models: DbModels, id: number) {
-    const plan = await models.plans.plan.getOne(id);
-
-    if (plan[0] === undefined) {
+  async findById(
+    models: Database,
+    id: number
+  ): Promise<{ id: PlanId; name?: string | null }> {
+    const plan = await models.plan.findOne({
+      where: { id: createBrandedValue(id) },
+    });
+
+    if (!plan) {
       throw new Error(`Plan with ID ${id} does not exist`);
     }
 
-    const planId = plan[0].id;
+    const planId = plan.id;
 
-    const latestPlanVersion = (await models.plans.planVersion.table().where({
-      planId: planId,
-      latestVersion: true,
-    })) as any as PlanVersionTable[];
+    const latestPlanVersion = await models.planVersion.findOne({
+      where: {
+        planId,
+        latestVersion: true,
+      },
+    });
```

Comment:
If this endpoint is unauthenticated, we instead want to get the `current` plan, to prevent inadvertently allowing access to unpublished plan data. Beyond that, if there is no `currentPlanVersion` that matches the requirements, we should 404 instead.

Reply:
It was like that when i got here :smile: Thanks for catching this. We now need https://github.com/UN-OCHA/hpc-api-core/pull/49 for 404 errors